src/nut_configurator.cc

```diff
@@ -145,6 +145,20 @@ void NUTConfigurator::systemctl( const std::string &operation, const std::string
     }
 }
 
+void NUTConfigurator::upsd_reload()
+{
+    std::vector<std::string> _argv = {"sudo", "/sbin/upsd", "-c", "reload" };
+    SubProcess upsd_c( _argv );
+    if( upsd_c.run() ) {
+        int result = upsd_c.wait();
+        log_info("sudo upsd -c reload result: %i (%s)",
+                 result,
+                 (result == 0 ? "ok" : "failed"));
```

Comment:
Note that this would fail if for any reason the `upsd` is not yet running (or not found via PID file). Maybe fall back to `systemctl restart nut-server` if direct reload failed?

Reply:
good catch !